File: src/test/java/leets/leenk/domain/feed/domain/service/FeedSaveServiceTest.java
package leets.leenk.domain.feed.domain.service;

import leets.leenk.domain.feed.domain.entity.Feed;
import leets.leenk.domain.feed.domain.repository.FeedRepository;
import leets.leenk.domain.feed.test.FeedTestFixture;
import leets.leenk.domain.feed.test.UserTestFixture;
import leets.leenk.domain.user.domain.entity.User;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtendWith;
import org.mockito.InjectMocks;
import org.mockito.Mock;
import org.mockito.junit.jupiter.MockitoExtension;

import static org.mockito.BDDMockito.then;

@ExtendWith(MockitoExtension.class)
public class FeedSaveServiceTest {
    @Mock
    private FeedRepository feedRepository;

    @InjectMocks
    private FeedSaveService feedSaveService;

    @Test
    @DisplayName("feedSave 테스트")
    void feedSave() {
        // given
        User user = UserTestFixture.createUser(1L, "me");
        Feed feed = FeedTestFixture.createFeed(null, user);

        // when
        feedSaveService.save(feed);

        // then
        then(feedRepository).should().save(feed);
    }
}

File: src/test/kotlin/leets/leenk/domain/feed/domain/service/FeedSaveServiceTest.kt
package leets.leenk.domain.feed.domain.service

import io.kotest.core.spec.style.BehaviorSpec
import io.mockk.every
import io.mockk.mockk
import io.mockk.verify
import leets.leenk.domain.feed.domain.repository.FeedRepository
import leets.leenk.domain.feed.test.FeedTestFixture
import leets.leenk.domain.feed.test.UserTestFixture

class FeedSaveServiceTest :
    BehaviorSpec({
        val feedRepository = mockk<FeedRepository>(relaxed = true)
        val feedSaveService = FeedSaveService(feedRepository)

        Given("사용자와 피드가 주어졌을 때") {
            val user = UserTestFixture.createUser(1L, "me")
            val feed = FeedTestFixture.createFeed(1L, user)

            every { feedRepository.save(any()) } returns feed

            When("피드를 저장하면") {
                feedSaveService.save(feed)

                Then("feedRepository의 save 메서드가 호출되어야 한다") {
                    verify { feedRepository.save(feed) }
                }
            }
        }
    })
